### PR TITLE
Sphinx-py: update to newer upstream version (1.8.1)

### PR DESCRIPTION
**Pull request by kamischi**
add py37 variant
add new package sphinxcontrib-websupport-py.
add dependencies and test-dependencies as they show up during the build.

### CODE REVIEW

**Comment by kamischi** [comment]
I edited the comment above. Is that OK?

**Comment by schwehr** [comment]
Yes, but you now have 3 different tenses in your comment. Please use:  "add"  to match the "update" verb of your commit summary

**Comment by kamischi** [comment]
Corrected. You've hit my weak spot in English on the head.

**Comment by danielj7** [comment]
Unfortunately, the py37 variant doesn't build because html5lib-py is missing a py37 variant. I tried updating it but it needs lxml-py37 which is also missing.

**Comment by danielj7** [comment]
I added html5lib-py37 but now packaging-py37 is still missing.

**Comment by kamischi** [comment]
Thanks for helping out. I will have another go at within a week.

**Comment by danielj7** [comment]
I submitted https://github.com/fink/fink-distributions/pull/298 to fix the missing dep.

**Comment by danielj7** [comment]
Ok, packaging-py update is merged, now pytz-py needs py37. PR https://github.com/fink/fink-distributions/pull/282

**Comment by dhomeier** [comment]
Looks good to me once #282 (and perhaps #299) are merged, apart from the following:

1. `alabaster-py` has become a package of its own and thus needs to go into `Depends` instead of building it as an extra source (should have updated this in my old Sphinx branch).

2. You may want to update the other auxiliary sources as well (especially `imagesize`, having jumped to a 1.x version):
```
Source2: https://files.pythonhosted.org/packages/source/i/imagesize/imagesize-1.1.0.tar.gz
Source2-MD5: 2f89749b05e07c79c46330dbc62f1e02
Source3: https://files.pythonhosted.org/packages/source/s/sphinx_rtd_theme/sphinx_rtd_theme-0.4.2.tar.gz
Source3-MD5: 524ba614e60bf214b621bc259f283ccd
```

3. For the `sphinxcontrib-websupport-py` `TestDepends` the `py37` version of `xapian30-py` has to be added yet; however in maintainer mode this will create another circular dependency, since xapian in turn depends on Sphinx...

I've tested this on 10.12.6.

File: 10.9-libcxx/stable/main/finkinfo/libs/pythonmods/sphinxcontrib-websupport-py.info [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    tox-py%type_pkg[python],
+    whoosh-py%type_pkg[python],
+    (%type_pkg[python]  = 27) xapian22-py%type_pkg[python],
+    (%type_pkg[python] >= 34) xapian30-py%type_pkg[python]
+  <<
+  TestScript: %p/bin/python%type_raw[python] setup.py test || exit 2
+  # The test above does nothing, as does nosetests.  The following trials
+  # also fail and i have no clue how to fix it.
+  # - %p/bin/python%type_raw[python] -m pytest
+  # - tox
```
Comment (nieder) [inline 10.9-libcxx/stable/main/finkinfo/libs/pythonmods/sphinxcontrib-websupport-py.info]: Have you tried `%p/bin/python%type_raw[python] -m pytest tests/ || exit 2` for running the tests?

**Comment by kamischi** [comment]
Tests still fail.

**Comment by nieder** [comment]
Opened sphinx-doc/sphinxcontrib-websupport#19 for the test failures.

I also tried a whole bunch of combos of test commands like @dhomeier did both with and without sphinx and sphinxcontrib-websupport installed, and they all failed one way or another. It just seems that the sphinxcontrib-websupport tests are broken and given the low number of commits upstream, doesn't look promising.

**Comment by nieder** [comment]
sphinx-doc/sphinx#4710 has a solution for the websupport failures inside sphinx-py. I took their fix into account, and I have sphinx-py27 failing just a single test. sphinxcontrib-websupport still fails a bunch, but almost fixing sphinx is a major victory.

**Comment by nieder** [comment]
The sole sphinx test failure I have left is due to my ISP resolving a dummy URL when it should return an invalid.

So all tests pass for me with my sphinx-py-1.8.2. sphinxcontrib-websupport still fails though.

@kamischi can I update this to have my sphinx-py-1.8.2?

**Comment by nieder** [comment]
Do setuptools-tng and babel-py need to be Depends for sphinx-py? They're already in BuildDepends. According to Ubuntu, they're BuildDepends.

**Comment by dhomeier** [comment]
Looks like you are right about setuptools-tng, but not babel-py:
```
File "/sw/bin/sphinx-build3.6", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/sw/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3105, in <module>
    @_call_aside
  File "/sw/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3089, in _call_aside
    f(*args, **kwargs)
  File "/sw/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3118, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/sw/lib/python3.6/site-packages/pkg_resources/__init__.py", line 578, in _build_master
    ws.require(__requires__)
  File "/sw/lib/python3.6/site-packages/pkg_resources/__init__.py", line 895, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/sw/lib/python3.6/site-packages/pkg_resources/__init__.py", line 781, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'babel!=2.0,>=1.3' distribution was not found and is required by Sphinx
```

**Comment by nieder** [comment]
The sphinxcontrib-websupport failures I had were caused by using pytest-py-4.0.1 because that's what tox autodownloads when running tests. Upstream sphinxcontrib-websupport just fixed that in sphinx-doc/sphinxcontrib-websupport#19, but since we don't do downloading during tests, running our pytest-3.6.4 directly instead leads to a bunch of test failures in sqlalchemy code (prob similar things @dhomeier saw earlier). Updating to sqlalchemy-1.2.15 didn't help.

**Comment by nieder** [comment]
yay on sphinxcontrib-websupport!
````python
/sw/bin/python2.7 -m pytest tests/ -v || exit 2
============================================================================ test session starts =============================================================================
platform darwin -- Python 2.7.15, pytest-3.6.4, py-1.5.4, pluggy-0.7.1 -- /sw/bin/python2.7
cachedir: .pytest_cache
rootdir: /sw/build.build/sphinxcontrib-websupport-py27-1.1.0-1/sphinxcontrib-websupport-1.1.0, inifile:
plugins: mock-1.10.0, expect-1.1.0
collected 13 items                                                                                                                                                           

tests/test_searchadapters.py::test_xapian PASSED                                                                                                                       [  7%]
tests/test_searchadapters.py::test_whoosh PASSED                                                                                                                       [ 15%]
tests/test_websupport.py::test_no_srcdir PASSED                                                                                                                        [ 23%]
tests/test_websupport.py::test_build PASSED                                                                                                                            [ 30%]
tests/test_websupport.py::test_get_document PASSED                                                                                                                     [ 38%]
tests/test_websupport.py::test_comments PASSED                                                                                                                         [ 46%]
tests/test_websupport.py::test_user_delete_comments PASSED                                                                                                             [ 53%]
tests/test_websupport.py::test_moderation PASSED                                                                                                                       [ 61%]
tests/test_websupport.py::test_moderator_delete_comments PASSED                                                                                                        [ 69%]
tests/test_websupport.py::test_update_username PASSED                                                                                                                  [ 76%]
tests/test_websupport.py::test_proposals PASSED                                                                                                                        [ 84%]
tests/test_websupport.py::test_voting PASSED                                                                                                                           [ 92%]
tests/test_websupport.py::test_differ PASSED                                                                                                                           [100%]

========================================================================= 13 passed in 6.12 seconds ==========================================================================
````
I applied the upstream changes since the 1.1.0 release (so basically matching their HEAD) and now tests pass across all variants for both sphinx and sphinxcontrib-websupport.

**Comment by nieder** [comment]
@kamischi should I just edit the changes I did onto this pull request? Will keep the history and seems easier than merging two branches before merging onto master.

**Comment by kamischi** [comment]
@nieder: No objections against your editing. Go as you please.

**Comment by nieder** [comment]
Please test

**Comment by kamischi** [comment]
I tested sphinx-py and sphinxcontrib-websupport-py with all variants. lgtm.

**Pull request merged**